Makefile

```diff
@@ -82,7 +82,7 @@ clean: clean-dist
 	@rm -f .toree-dev-image
 	@-docker rmi -f $(TOREE_DEV_IMAGE)
 
-.toree-dev-image:
+.toree-dev-image: .clean-toree-dev-image
```

Comment:
without this change, `make .toree-dev-image` will be skipped after the first time run.

the change could be verified via

```
make .clean-toree-dev-image
make .toree-dev-image
make .toree-dev-image
```

Before:
```
(toree) ➜  apache-toree git:(master) make .clean-toree-dev-image
Untagged: apache/toree-dev:latest
Deleted: sha256:d9bf8641a323aa98d66dcacda802ad1c535f0041a23a51a59f352255122fa96f
(toree) ➜  apache-toree git:(master) make .toree-dev-image      
[+] Building 0.0s (13/13) FINISHED                                                                                                                                                           docker:orbstack
 => [internal] load build definition from Dockerfile.toree-dev                                                                                                                                          0.0s
 => => transferring dockerfile: 2.32kB                                                                                                                                                                  0.0s
 => [internal] load .dockerignore                                                                                                                                                                       0.0s
 => => transferring context: 2B                                                                                                                                                                         0.0s
 => [internal] load metadata for docker.io/jupyter/all-spark-notebook:latest                                                                                                                            0.0s
 => [1/8] FROM docker.io/jupyter/all-spark-notebook                                                                                                                                                     0.0s
 => [internal] load build context                                                                                                                                                                       0.0s
 => => transferring context: 1.18kB                                                                                                                                                                     0.0s
 => CACHED [2/8] RUN apt-get -y update &&     apt-get install -y --no-install-recommends openjdk-8-jdk ca-certificates-java  &&     apt-get clean &&     rm -rf /var/lib/apt/lists/* &&     update-ca-  0.0s
 => CACHED [3/8] RUN cd /tmp &&     wget -q https://archive.apache.org/dist/spark/spark-3.3.2/spark-3.3.2-bin-hadoop3.tgz &&     tar xzf spark-3.3.2-bin-hadoop3.tgz -C /usr/local &&     rm spark-3.3  0.0s
 => CACHED [4/8] RUN cd /usr/local &&     rm spark &&     ln -s spark-3.3.2-bin-hadoop3 spark                                                                                                           0.0s
 => CACHED [5/8] RUN cd /opt/conda/share/jupyter/kernels/ &&     rm -rf spylon-kernel                                                                                                                   0.0s
 => CACHED [6/8] ADD dist/toree /usr/local/share/jupyter/kernels/toree                                                                                                                                  0.0s
 => CACHED [7/8] ADD etc/kernel.json /usr/local/share/jupyter/kernels/toree/kernel.json                                                                                                                 0.0s
 => CACHED [8/8] RUN fix-permissions /usr/local/share/jupyter/                                                                                                                                          0.0s
 => exporting to image                                                                                                                                                                                  0.0s
 => => exporting layers                                                                                                                                                                                 0.0s
 => => writing image sha256:d9bf8641a323aa98d66dcacda802ad1c535f0041a23a51a59f352255122fa96f                                                                                                            0.0s
 => => naming to docker.io/apache/toree-dev                                                                                                                                                             0.0s
touch .toree-dev-image
(toree) ➜  apache-toree git:(master) make .toree-dev-image
make: `.toree-dev-image' is up to date.
```

After:
```
(toree) ➜  apache-toree git:(TOREE-553) make .clean-toree-dev-image
Untagged: apache/toree-dev:latest
Deleted: sha256:d9bf8641a323aa98d66dcacda802ad1c535f0041a23a51a59f352255122fa96f
(toree) ➜  apache-toree git:(TOREE-553) make .toree-dev-image      
Error response from daemon: No such image: apache/toree-dev:latest
[+] Building 0.0s (13/13) FINISHED                                                                                                                                                           docker:orbstack
 => [internal] load build definition from Dockerfile.toree-dev                                                                                                                                          0.0s
 => => transferring dockerfile: 2.32kB                                                                                                                                                                  0.0s
 => [internal] load .dockerignore                                                                                                                                                                       0.0s
 => => transferring context: 2B                                                                                                                                                                         0.0s
 => [internal] load metadata for docker.io/jupyter/all-spark-notebook:latest                                                                                                                            0.0s
 => [1/8] FROM docker.io/jupyter/all-spark-notebook                                                                                                                                                     0.0s
 => [internal] load build context                                                                                                                                                                       0.0s
 => => transferring context: 1.18kB                                                                                                                                                                     0.0s
 => CACHED [2/8] RUN apt-get -y update &&     apt-get install -y --no-install-recommends openjdk-8-jdk ca-certificates-java  &&     apt-get clean &&     rm -rf /var/lib/apt/lists/* &&     update-ca-  0.0s
 => CACHED [3/8] RUN cd /tmp &&     wget -q https://archive.apache.org/dist/spark/spark-3.3.2/spark-3.3.2-bin-hadoop3.tgz &&     tar xzf spark-3.3.2-bin-hadoop3.tgz -C /usr/local &&     rm spark-3.3  0.0s
 => CACHED [4/8] RUN cd /usr/local &&     rm spark &&     ln -s spark-3.3.2-bin-hadoop3 spark                                                                                                           0.0s
 => CACHED [5/8] RUN cd /opt/conda/share/jupyter/kernels/ &&     rm -rf spylon-kernel                                                                                                                   0.0s
 => CACHED [6/8] ADD dist/toree /usr/local/share/jupyter/kernels/toree                                                                                                                                  0.0s
 => CACHED [7/8] ADD etc/kernel.json /usr/local/share/jupyter/kernels/toree/kernel.json                                                                                                                 0.0s
 => CACHED [8/8] RUN fix-permissions /usr/local/share/jupyter/                                                                                                                                          0.0s
 => exporting to image                                                                                                                                                                                  0.0s
 => => exporting layers                                                                                                                                                                                 0.0s
 => => writing image sha256:d9bf8641a323aa98d66dcacda802ad1c535f0041a23a51a59f352255122fa96f                                                                                                            0.0s
 => => naming to docker.io/apache/toree-dev                                                                                                                                                             0.0s
touch .toree-dev-image
(toree) ➜  apache-toree git:(TOREE-553) make .toree-dev-image
Untagged: apache/toree-dev:latest
Deleted: sha256:d9bf8641a323aa98d66dcacda802ad1c535f0041a23a51a59f352255122fa96f
[+] Building 0.0s (13/13) FINISHED                                                                                                                                                           docker:orbstack
 => [internal] load build definition from Dockerfile.toree-dev                                                                                                                                          0.0s
 => => transferring dockerfile: 2.32kB                                                                                                                                                                  0.0s
 => [internal] load .dockerignore                                                                                                                                                                       0.0s
 => => transferring context: 2B                                                                                                                                                                         0.0s
 => [internal] load metadata for docker.io/jupyter/all-spark-notebook:latest                                                                                                                            0.0s
 => [1/8] FROM docker.io/jupyter/all-spark-notebook                                                                                                                                                     0.0s
 => [internal] load build context                                                                                                                                                                       0.0s
 => => transferring context: 1.18kB                                                                                                                                                                     0.0s
 => CACHED [2/8] RUN apt-get -y update &&     apt-get install -y --no-install-recommends openjdk-8-jdk ca-certificates-java  &&     apt-get clean &&     rm -rf /var/lib/apt/lists/* &&     update-ca-  0.0s
 => CACHED [3/8] RUN cd /tmp &&     wget -q https://archive.apache.org/dist/spark/spark-3.3.2/spark-3.3.2-bin-hadoop3.tgz &&     tar xzf spark-3.3.2-bin-hadoop3.tgz -C /usr/local &&     rm spark-3.3  0.0s
 => CACHED [4/8] RUN cd /usr/local &&     rm spark &&     ln -s spark-3.3.2-bin-hadoop3 spark                                                                                                           0.0s
 => CACHED [5/8] RUN cd /opt/conda/share/jupyter/kernels/ &&     rm -rf spylon-kernel                                                                                                                   0.0s
 => CACHED [6/8] ADD dist/toree /usr/local/share/jupyter/kernels/toree                                                                                                                                  0.0s
 => CACHED [7/8] ADD etc/kernel.json /usr/local/share/jupyter/kernels/toree/kernel.json                                                                                                                 0.0s
 => CACHED [8/8] RUN fix-permissions /usr/local/share/jupyter/                                                                                                                                          0.0s
 => exporting to image                                                                                                                                                                                  0.0s
 => => exporting layers                                                                                                                                                                                 0.0s
 => => writing image sha256:d9bf8641a323aa98d66dcacda802ad1c535f0041a23a51a59f352255122fa96f                                                                                                            0.0s
 => => naming to docker.io/apache/toree-dev                                                                                                                                                             0.0s
touch .toree-dev-image
(toree) ➜  apache-toree git:(TOREE-553) 

```

Reply:
I am not sure I understand this change, what you describe above is by design, you can keep making `make dev` without having to rebuild the docker image assuming the toree code is attached to the image via volumes. If you need to rebuild the docker image in itself, `make clean dev` should do it.

Reply:
@lresende from my understanding

1. The dev image uses the command `ADD` instead of volume to ship the dist artifacts, artifacts won't update without rebuilding the image

https://github.com/apache/incubator-toree/blob/937c1260bf7973ef20b0052989398ac9e789d160/Dockerfile.toree-dev#L52

2. `make clean` won't trigger `.clean-toree-dev-image`, additionally, `.binder-image` also depends on `.clean-binder-image`.

https://github.com/apache/incubator-toree/blob/937c1260bf7973ef20b0052989398ac9e789d160/Makefile#L89-L95

Reply:
@lresende appreciate it if you could take another look

Reply:
@lresende would you mind taking another look at my last comment?